### PR TITLE
Shard plugins for example app builds

### PR DESCRIPTION
**Pull request by mravn-google**
Make use of sharding parameters in `flutter_plugin_tools`, with shard count = 2.

My .travis.yaml fu is obviously weak, and I have not yet had time to find a better approach that has less duplication of configuration and doesn't result in two shards also when there is only one plugin involved in the CI run.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -60,51 +61,57 @@ matrix:
...
-        - gradle -v
-        - git clone https://github.com/flutter/flutter.git
         - export PATH=`pwd`/flutter/bin:`pwd`/flutter/bin/cache/dart-sdk/bin:$PATH
-        - flutter doctor
-        - pub global activate flutter_plugin_tools
       script:
         - ./script/plugin_tools.sh build-examples --apk
         - ./script/plugin_tools.sh java-test  # must come after apk build
-    # Job 4) Build example IPAs
+    # Job 3.2) Build example APKs and run Java tests, shard 2/2
```
Comment (mravn-google) [inline .travis.yml]: I would have liked to avoid duplication here, but not sure what is the best approach for that.

File: script/plugin_tools.sh [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
   FLUTTER_CHANGED_GLOBAL=`git diff --name-only $TRAVIS_COMMIT_RANGE | grep -v packages | wc -l`
   FLUTTER_CHANGED_PACKAGES=`git diff --name-only $TRAVIS_COMMIT_RANGE | grep -o "packages/[^/]*" | sed -e "s/packages\///g" | sort | uniq | paste -s -d, -`
   if [ "${FLUTTER_CHANGED_PACKAGES}" = "" ] || [ $FLUTTER_CHANGED_GLOBAL -gt 0 ]; then
-    echo "Running for all packages"
-    pub global run flutter_plugin_tools "$@"
+    echo "Running for all packages ($PLUGIN_SHARDING)"
+    pub global run flutter_plugin_tools "$@" $PLUGIN_SHARDING
   else
     echo "Running only for $FLUTTER_CHANGED_PACKAGES"
     pub global run flutter_plugin_tools "$@" --plugins=$FLUTTER_CHANGED_PACKAGES
```
Comment (fkorotkov) [inline script/plugin_tools.sh]: @mravn-google should `$PLUGIN_SHARDING` be added here as well?

Comment (mravn-google) [reply script/plugin_tools.sh]: That wasn't my original intention, as this branch usually means a small number of plug-ins. Trouble is, the Travis config spawns the same number of jobs either way, so we end up with redundant execution.

Comment (fkorotkov) [reply script/plugin_tools.sh]: Got it! I was just wondering if `$PLUGIN_SHARDING` was missed here.

**Pull request merged**